packages/@okta/vuepress-site/docs/guides/authenticators-google-authenticator/main/ios/enrollmentintegrationsteps.md

```diff
@@ -0,0 +1,550 @@
+    case .password:
+        remediation["credentials.passcode"]?.value = password
+        remediation.proceed()
+
+    default:
+        // Handle the case of an unknown authenticator.
+        return
+    }
+}
+
+```
+
+>**Note:** The response from the server specifies either the next step, an error, or a message, such as an incorrect code. The code that handles the response may need to show different UI, depending on the step. The `IDXClientDelegate` protocol includes two other functions: `idx(client:didReceive<Token>:)` handles receiving a sign-in token, and   `idx(client:didReceive<Error>:)` handles errors.
+
+### 4: Display a list of authenticators available for enrollment
+
+After verifying the username and password, the server sends a remediation of type `.selectAuthenticatorEnroll` that contains the possible authenticators. The list of choices are in the `options` property of a field with `name` set to `"authenticator"` in the remediation `forms` property.
+
+The display name of an authenticator is in the `label` property of the field. Use this name to build your selection UI. For an example of using `label`, see the code that displays the the `Picker` in the code for `AuthenticatorSelectionView` later in this step.
+
+When the user selects an option, set the `selectedOption` property of the field containing the options (the one with the `name` `"authenticator"`) to the field containing the selected option. Call `proceed()` on the remediation once the user has finalized their choice to send the information and request the next step.
+
+This utility function returns a tuple (`AuthenticatorOptions` type alias) with the field that contains the options, an array of options, and the selected field:
+
+
+```swift
+typealias AuthenticatorOptions = (optionSelectorField: Remediation.Form.Field,
+                                  optionFields: [Remediation.Form.Field],
+                                  selectedField: Remediation.Form.Field?)
+
+
+func authenticatorOptions() -> AuthenticatorOptions? {
+    guard let fields = currentRemediation?.form.fields,
+          let selectorField = fields.filter({ $0.name == "authenticator"}).first,
+          let options = selectorField.options,
+          options.count > 0 else {
+              return nil
+          }
+    return (selectorField, options, selectorField.selectedOption)
+}
+
+```
+
+The function that handles the remediation type checks that there's at least one option:
+
+```swift
+func handleSelectAuthenticatorToEnroll(_ remediation: Remediation) {
+    guard let optionsInfo = authenticatorOptions(),
+          optionsInfo.optionFields.count > 0 else {
+              finish(with: .unexpectedAuthenticator)
+              return
+          }
+
+    // Tell the caller to present a UI for choosing an authenticator.
+}
+```
+
+Use the information returned by these utility functions to populate a UI for selecting an authenticator.If there are no options or some other issue, present an error.
```

Comment:
```suggestion
Use the information returned by these utility functions to populate a UI for selecting an authenticator. If there are no options or some other issue, present an error.
```

Reply:
Another nice catch. Fixed in the next commit.